File: ADIP/sdks/typescript/examples/quickstart.ts
/**
 * ADIP TypeScript SDK — quickstart example.
 * Run the backend first, then: npx tsx examples/quickstart.ts
 */
import { ADIPClient, ADIPError } from "../src/index.js";

async function main(): Promise<void> {
  const client = new ADIPClient({ baseUrl: "http://localhost:8000" });
  try {
    console.log("Health:", await client.health());

    const projects = (await client.listProjects(1, 5)) as { items?: Array<{ id: number }> };
    const items = projects.items ?? [];
    console.log(`Projects: ${items.length}`);
    if (items.length === 0) return;
    const pid = items[0].id;

    const result = (await client.orchestrate(
      "Implement UPI Auto-Reversal with NPCI reconciliation.",
    )) as { phases?: unknown[] };
    console.log("Orchestration phases:", result.phases?.length ?? 0);

    const artifact = (await client.generateArtifact(pid, "BRD")) as {
      reference: string;
      sections: unknown[];
    };
    console.log("Artifact:", artifact.reference, "sections:", artifact.sections.length);

    const reg = (await client.regressionCompare(
      "Implement UPI reversal.",
      "Implement UPI reversal with reconciliation & audit.",
    )) as { overall_verdict: string };
    console.log("Regression verdict:", reg.overall_verdict);
  } catch (err) {
    if (err instanceof ADIPError) {
      console.error(`ADIP error: ${err.message} (status=${err.status})`);
    } else {
      throw err;
    }
  }
}

main();

File: ADIP/sdks/python/examples/quickstart.py
"""ADIP Python SDK — quickstart example.

Run the backend first (uvicorn app.main:app), then:  python quickstart.py
"""
from adip_sdk import ADIPClient, ADIPError


def main() -> None:
    client = ADIPClient("http://localhost:8000")

    try:
        print("Health:", client.health())
        print("LLM health:", client.llm_health())

        projects = client.list_projects(size=5)
        items = projects.get("items", projects)
        print(f"Projects: {len(items)}")
        if not items:
            return
        pid = items[0]["id"]

        # One prompt → full SDLC
        result = client.orchestrate("Implement UPI Auto-Reversal with NPCI reconciliation.")
        print("Orchestration phases:", len(result.get("phases", [])))

        # Generate + score + export an artifact
        artifact = client.generate_artifact(pid, "BRD")
        print("Artifact:", artifact["reference"], "sections:", len(artifact["sections"]))
        quality = client.score_artifact_quality(pid, "BRD")
        print("Quality:", quality.get("overall_score"), quality.get("quality_band"))
        md = client.export_artifact(pid, "BRD", "markdown")
        print("Markdown export chars:", len(md))

        # Benchmark + regression
        bench = client.benchmark(["Implement UPI reversal.",
                                  "Implement UPI reversal with reconciliation & audit."])
        print("Best benchmark version:", bench.get("best_version"))
        reg = client.regression_compare("Implement UPI reversal.",
                                        "Implement UPI reversal with reconciliation & audit.")
        print("Regression verdict:", reg.get("overall_verdict"))

    except ADIPError as exc:
        print(f"ADIP error: {exc} (status={exc.status})")


if __name__ == "__main__":
    main()
